File: easy/0001_two_sum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> um;
        vector<int> result;

        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (um.find(complement) != um.end()) {
                result.push_back(i);
                result.push_back(um[complement]);
                return result;
            }
            um[nums[i]] = i;
        }

        return result;
    }    
};

File: neetcode/arrays-hashing/0001-two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        freq = {}
        for i in range(len(nums)):
            look_for = target - nums[i]

            if look_for in freq:
                return [i, freq[look_for]]
            
            freq[nums[i]] = i
            
        return []